packages/loader/container-loader/src/container.ts

```diff
@@ -1003,6 +1013,9 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     private disconnectInternal() {
         assert(!this.closed, 0x2c7 /* "Attempting to disconnect() a closed Container" */);
 
+        // Send abort signal to cancel any in-progress connect attempts
+        this.connectAbortController.abort();
```

Comment:
Cancelation is hard. One thing that is not accounted here - asynchrony of cancelation.
Setting it to cancelled starts the chain of events, but it may take a while to finish cancelation process (especially in first iteration, where we do not propagate cancel signal up to socket.io).
As result, a new connect() request will go to void, as connectCore() will ignore it (it still has pending connection).

I see multiple approached here:
1. We ensure that connect() request that comes right after cancelation will reset IProgress object in DeltaManager, and thus would allow pending connection to go through when it completes.
2. We make cancelation blocking and wait until it's fully unrolls. This, plus serialization of all connect() & disconnect() calls make it safe (but ugly, and not performant in presence of asynchrony).
3. We find a way to package connectivity machinery into its own object, and disconnect() request would throw away this object (i.e. get rid of any reference to it). Cancelation may still take time, but it does not impact global state - next request to connect() would simply start a new request (a bit of extra resources, but eventually we will make it cheaper by ensuring socket.io is cancelled faster).

It would be great to cover it in UT - rapid disconnect / connect flow with actual connection taking some time.